torch/csrc/jit/passes/quantization.cpp

```diff
@@ -969,7 +982,7 @@ graph(%self, %scale, %zero_point, %dtype):
     %weight_quant = prim::GetAttr[name="_quantized_weight"](%self)
     return (%weight_quant) )";
   SubgraphRewriter rewriter;
-  rewriter.RegisterRewritePattern(pattern, replacement);
+  rewriter.RegisterRewritePattern(pattern.pattern_string, replacement);
```

Comment:
Here we would actually reparse the string in subgraph-rewriter, we probably should not be changing this place. And if we want to cache this parsing, we should just keep the rewriter object alive.

Reply:
The redundant parsing is an issue whether we cache the results or not. I'm going to go ahead and cache the SubgraphRewriter. That way, there's still a redundant parse, but all of the work is memoized

Reply:
What I'm saying is that this particular change only makes it worse: we now parse it when for memoization **and** in subgraph-rewriter.

Also I think using subgraph rewriter here is suboptimal anyway and we should just use the matcher directly - in fact we're already doing it with the same pattern just 10 lines earlier. The only thing it is used for here is to remove one node from the matched subgraph - just do it directly in the loop above.

Reply:
@ZolotukhinM before this change, we had one `parseIR` call at line 936, and one `parseIR` call embedded in the `runOnGraph` call at line 973 (by way of `runOnGraph -> rewriteSinglePatternOnGraph ->parseIR`). These two callsites were hit every time `FoldQuantizeCallIntoBuffer` was called. My change makes it so that the first such call is done once and stored away, leaving only the second call within `runOnGraph`.

I agree with your second point, I can just switch this over and be done with it

Reply:
Right, makes sense. I misunderstood your earlier point and thought you were claiming savings in subgraph-rewriter itself.